Sources/SocksCore/Conversions.swift

```diff
@@ -20,9 +20,30 @@ extension Array {
 }
 
 extension String {
-    
     public func toBytes() -> [UInt8] {
         return Array(self.utf8)
     }
 }
 
+// Sendable and Receivable implementations
+
+extension String: Sendable {
+    public func serialize() -> [UInt8] {
+        return self.toBytes()
+    }
+}
+
+extension String: Receivable {
+    static func deserialize(reader: (maxBytes: Int) throws -> [UInt8]) throws -> String {
```

Comment:
@LoganWright So you'd be interested in this feature? That's great!

1. I dislike the lack of symmetry of `serialize/parse`, especially since there's a word opposite to `serialize` (`deserialize`). Maybe something simpler like `fromData` and `toData` might do?

2. `@noescape` would make a guarantee that parsing will happen synchronously, even for new objects that will implement this protocol later. I'm not sure if we want to make that.